src/main/java/org/reactome/server/service/controller/citation/CitationController.java

```diff
@@ -0,0 +1,294 @@
+
+        List<String> urls = new ArrayList<>();
+        if(map.containsKey("doi")){
+            String doi = (String)map.get("doi");
+            citation.setDoi(doi);
+            urls.add(DOI_BASE_URL + doi);
+        }
+        else if(map.containsKey("stid")) {
+            urls.add(map.get("baseURL") + "/content/detail/" + map.get("stid"));
+        }
+        citation.setUrls(urls);
+
+        if(map.containsKey("authorList")) {
+            Map<String, Object> authorList = (Map<String, Object>) map.get("authorList");
+            citation.setAuthors(authorList.containsKey("author") ? (List<Map<String, String>>)authorList.get("author") : null);
+        }
+        else if(map.containsKey("authors")) {
+            citation.setAuthors((List<Map<String, String>>)map.get("authors"));
+        }
+
+        citation.setPathway((Boolean)map.get("isPathway"));
+        return citation;
+    }
+
+    // private method for getting the citation details for a static citation
+    // for export purposes
+    private Map<String, Object> getStaticCitationForExport(String id) {
+        try {
+            HttpGet request = new HttpGet(new URIBuilder()
+                    .setScheme("https")
+                    .setHost("www.ebi.ac.uk")
+                    .setPath("/europepmc/webservices/rest/search")
+                    .setParameter("query", id)
+                    .setParameter("format", "json")
+                    .setParameter("resultType", "core")
+                    .build());
+
+            try (CloseableHttpClient httpClient = HttpClients.createDefault();
+                 CloseableHttpResponse response = httpClient.execute(request)) {
+
+                // if the EuropePMC API is down or response is not okay for whatever reason...
+                if (response.getStatusLine().getStatusCode() != HttpStatus.OK.value()) {
+                    return null;
+                }
+
+                else {
+                    String jsonResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    return objectMapper.convertValue(objectMapper.readTree(jsonResponse).get("resultList").get("result").get(0), new TypeReference<Map<String, Object>>(){});
+                }
+            }
+            catch (Exception e) {
+                throw new Exception(e);
+            }
+        }
+        catch (Exception e) {
+            infoLogger.error("Exception thrown in getStaticCitationForExport method", e);
+            return null;
+        }
+    }
+
+    // helper function
+    public String getURLBase(HttpServletRequest request) throws MalformedURLException {
+        URL requestURL = new URL(request.getRequestURL().toString());
+        String port = requestURL.getPort() == -1 ? "" : ":" + requestURL.getPort();
+        return requestURL.getProtocol() + "://" + requestURL.getHost() + port;
```

Comment:
Not really sure about this function. It doesn't make sense to cite dev.reactome or any other server, we should always cite based in reactome.org. 
If I'm missing something, what's the purpose of this function ?